### PR TITLE
Increase audio device polling time to 1500ms

### LINKED ISSUE #119: [OS X] SingStar (wireless) microphones not recognized on (my?) Mac
This has been a show stopper for productively using (a.k.a. singing) UltraStar deluxe on my Mac for a while and I can't figure out if it is just a problem on my Mac or if it is a general problem.

If I connect my SingStar wireless microphones to my Mac, they are properly detected by the system (checked in Settings --> Sound):

bildschirmfoto 2016-08-14 um 12 38 24

but in the UltraStar deluxe settings, all I see is the internal microphone:

bildschirmfoto 2016-08-14 um 12 39 48

I appreciate any help with this regard. Let me know if you have any ideas...

### PR DESCRIPTION
**Pull request by taligentx**
Polling on Mac OS X 10.11 seems to be taking between 450-1100ms. Fixes #119

### CODE REVIEW

**Comment by basisbit** [comment]
can anyone confirm that thi actually fixes (workaround) the issue?

**Comment by bohning** [comment]
If the change is already in the master branch, I can check later tonight

**Comment by taligentx** [comment]
@bohning It's not merged yet pending confirmation that it works for somebody besides myself - if you'd like you can pull from https://github.com/taligentx/USDX to test and confirm if it works for you.

**Comment by bohning** [comment]
Works like a charm! Awesome! (Note the text artefacts at the top of the screenshot though...)
bildschirmfoto 2016-09-15 um 19 50 17

**Comment by AlexanderS** [comment]
You can also directly pull the pull requests: https://help.github.com/articles/checking-out-pull-requests-locally/

So something like this should work:

```
git checkout -b test
git fetch origin pull/128/head
git merge FETCH_HEAD
```

**Comment by bohning** [comment]
@AlexanderS: unfortunately saw your hint too late, did a full checkout of the taligentix fork...

**Comment by basisbit** [comment]
@bohning these artefacts happen when you use an old theme. replace the theme files and it will be fixed.

**Comment by bohning** [comment]
@basisbit Okay, thanks for the hint. Where do I get the new theme files? They are not in the repository?

**Comment by RattleSN4K3** [comment]
> Where do I get the new theme files? They are not in the repository?

The themes are in the repo and also updated. _Deluxe_ and _Modern_ work just fine. One thing you can see from your screenshot, _**Threshold**_ and _**Mic boost**_ options are missing.

screenshot0172

These options were previously in another option category but now moved to _Record_ since it's related to microphone input. The artefacts you see are uncustomized default buttons/sliders located at (0,0).

**Comment by bohning** [comment]
Sorry for being such a noob, but I don't know how to fix this. If they are in the repository, the should be updated with every update I get (I use git desktop on the Mac). The theme I have selected is Deluxe. What do I need to do to fix this? Thanks for your help.

**Comment by RattleSN4K3** [comment]
Is it the same for a clean cloned repo? Otherwise, not sure if it is even possible, but are you updating the root and not just `src/`?

The Deluxe theme should have both options in the ini file Deluxe.ini
``` ini
[OptionsRecordSelectThreshold]
Tex = MainBar
Type = Transparent
TexSBG = SelectBG
TypeSBG = Transparent
Text = SING_OPTIONS_SOUND_THRESHOLD
X = 70
Y = 367
W = 250
H = 40
SkipX = 10

DColor = ColorDark
Color = ColorLight
TColor = White
TDColor = White
 
SBGDColor = ColorDark
SBGColor = ColorLight
STColor = White
STDColor = GrayDark

[OptionsRecordSelectMicBoost]
Tex = MainBar
Type = Transparent
TexSBG = SelectBG
TypeSBG = Transparent
Text = SING_OPTIONS_SOUND_MIC_BOOST
X = 70
Y = 417
W = 250
H = 40
SkipX = 10

DColor = ColorDark
Color = ColorLight
TColor = White
TDColor = White
 
SBGDColor = ColorDark
SBGColor = ColorLight
STColor = White
STDColor = GrayDark
```

Both these items are loaded for the Record menu in the UThemes class:
https://github.com/UltraStar-Deluxe/USDX/blob/master/src/base/UThemes.pas#L2323
``` pascal
      // Options Record
      ThemeLoadBasic(OptionsRecord, 'OptionsRecord');

      ThemeLoadSelectSlide(OptionsRecord.SelectSlideCard,     'OptionsRecordSelectSlideCard');
      ThemeLoadSelectSlide(OptionsRecord.SelectSlideInput,    'OptionsRecordSelectSlideInput');
      ThemeLoadSelectSlide(OptionsRecord.SelectSlideChannel,  'OptionsRecordSelectSlideChannel');
      ThemeLoadSelectSlide(OptionsRecord.SelectThreshold,     'OptionsRecordSelectThreshold'); //basisbit TODO
      ThemeLoadSelectSlide(OptionsRecord.SelectMicBoost,      'OptionsRecordSelectMicBoost');
      ThemeLoadButton(OptionsRecord.ButtonExit, 'OptionsRecordButtonExit');
```

... and added to the screen in the create method of UScreenOptionsRecord:
https://github.com/UltraStar-Deluxe/USDX/blob/master/src/screens/UScreenOptionsRecord.pas#L344
``` pascal
    Theme.OptionsRecord.SelectThreshold.showArrows := true; //basisbit TODO
    Theme.OptionsRecord.SelectThreshold.oneItemOnly := true;
    SelectThresholdID := AddSelectSlide(Theme.OptionsRecord.SelectThreshold, Ini.ThresholdIndex, IThreshold);

    Theme.OptionsRecord.SelectMicBoost.showArrows := true;
    Theme.OptionsRecord.SelectMicBoost.oneItemOnly := true;
    AddSelectSlide(Theme.OptionsRecord.SelectMicBoost, Ini.MicBoost, IMicBoostTranslated);
```

So basically you should verify if you have these things in there, otherwise you'd need to update the repo properly.

**Comment by basisbit** [comment]
I'd rather guess that the install script for OS X does not replace the theme if it already exists in the user's folder.

**Comment by bohning** [comment]
Okay, this is what I do: I use the GitHub Desktop application, with two repositories: USDX and USDX_taligentix. Both of them I can update with the convenient 'Sync' button in the application.
Issuing git pull in the terminal confirms: Already up-to-date.

I then configure and build in the terminal via
sudo ./configure --enable-osx-brew
sudo make macosx-standalone-app

and start the resulting app via double click in Finder.
Alternatively, I can build the dmg disk image, mount the image and copy the app to programs.
Either way, I have the above behavior.
Any ideas? Thanks in advance.

**Comment by bohning** [comment]
bildschirmfoto 2016-09-16 um 18 55 40

**Comment by bohning** [comment]
Removing the '~/Library/ApplicationSupport/UltraStarDeluxe1.3' folder and reinstalling the app fixes the theme problem. Now I see 'Threshold' and 'Mic boost' correctly in the 'Record Options' as shown in the above screenshot of @RattleSN4K3.

Thanks!
